lightningd/pay.c

```diff
@@ -179,9 +179,36 @@ sendpay_result_in_progress(const tal_t *ctx,
 	return result;
 }
 
+/* Report to wallet nodestats that the specified nodes are
+ * alive. */
+static void
+report_nodestats_alive(struct wallet *wallet,
+		       const struct pubkey *nodes,
+		       int num_nodes)
```

Comment:
I think we're using a `tal_arr` here anyway, so we could use `tal_count`.

Reply:
ok

Reply:
On checking again, no.  On routing failure, we know that all nodes up to the erring node are alive (else we would not have received a routing failure onion reply), but not any of the succeeding nodes.  So we call this function with `num_nodes < tal_count(nodes)`.

We would not be able to receive a remote routing failure if for instance the first hop failed (we would be the node that raises the error to ourself, and would not call this code).  But actually receiving a remote routing failure means that every node on the route, up to the node that raised the error, is alive, so we should mark them as being seen as alive in our stats.